file-service-persistence/src/main/java/uk/gov/justice/services/fileservice/repository/ConnectionProvider.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.services.fileservice.repository;
+
+import static java.lang.String.format;
+
+import uk.gov.justice.services.fileservice.api.FileServiceException;
+import uk.gov.justice.services.jdbc.persistence.InitialContextFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import javax.enterprise.inject.Default;
+import javax.inject.Inject;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+/**
+ * Gets a hard JDBC {@link Connection} using a the container Datasource and JNDI
+ */
+@Default
+public class ConnectionProvider {
```

Comment:
I'm not convinced this should be a thing... From a naming perspective, a "provider" in JEE is something specific; it is for providing instances of something for injection. I would be okay with having a class that provides the DataSource, but it feels like the repositories themselves should be responsible for getting their connections from the datasource directly rather having that logic in here.